src/java.base/share/classes/java/lang/Long.java

```diff
@@ -549,27 +551,39 @@ static int getChars(long i, int index, byte[] buf) {
         // Get 2 digits/iteration using longs until quotient fits into an int
         while (i <= Integer.MIN_VALUE) {
             q = i / 100;
-            r = (int)((q * 100) - i);
+            charPos -= 2;
+            UNSAFE.putShortUnaligned(
+                    buf,
+                    Unsafe.ARRAY_BYTE_BASE_OFFSET + charPos,
+                    Integer.PACKED_DIGITS[(int)((q * 100) - i)],
+                    false);
```

Comment:
Add the array bound check for the store of the characters.

Reply:
I added assert with reference to the implementation of StringUTF16#putChar, is this safe enough?

Reply:
Can you use `VarHandle` here?